src/runner/main.nf

```diff
@@ -9,6 +9,16 @@ workflow run_wf {
 
   main:
     output_ch = input_ch
+      // Extract the ID from the input.
+      // If this is a directory or tar-gz file containing `.`'s,
+      // only select the part before the first `.`.
+      | map{ id, state ->
+        def id_with_dots = state.input.getFileName().toString()
```

Comment:
In the actual workflow we detect a tar file with `.tar.gz`, `.tar` and `.tgz`. I think we could just strip those three here and use that? This avoids problems with extra dots like `this.is.an.example.tar.gz`